### DISCUSSION TITLE
How to Cancel a step created from EmrAddStepsOperator from Airflow on EMR

### DISCUSSION THREAD

**Discussion by swapz-z** [apache/airflow#27959] [Q&A] [answered]
I created steps using EmrAddStepsOperator and wait for their state change using EmrStepSensor. 
How can I send a cancel step command from Airflow DAG for a particular step to EMR (AWS ) 

Would marking the step in DAG as SUCCESS/FAILURE/CLEAR help ?

**Comment by o-nikolas** [comment, marked as answer]
Marking the step as failed/clear would not cancel the running step in your AWS account.

There is a [cancel_steps](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/emr.html#EMR.Client.cancel_steps) api in boto, but there is not currently an Airflow operator for that usecase.

This is a reasonable feature request.

**Reply by o-nikolas** [reply]
Issue for feature request: https://github.com/apache/airflow/issues/27979
